### DISCUSSION TITLE
NFC Usage on Seeed Studio XIAO nRF52840 (Sense) | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#214] [Q&A]
# NFC Usage on Seeed Studio XIAO nRF52840 (Sense) | Seeed Studio Wiki

NFC Usage on Seeed Studio XIAO nRF52840 (Sense)

https://wiki.seeedstudio.com/XIAO-BLE-Sense-NFC-Usage/

**Comment by hasanbasri1993** [comment]
i getting error
Using Arduino IDE 2
macOS 13.5

fatal error: NFC.h: No such file or directory
    1 | #include <NFC.h>

**Reply by MatthewJeffson** [reply]
Hi, the NFC library is currently under updating and will be released as soon as possible.

**Reply by hasanbasri1993** [reply]
Thanks u for your reply
Hope u add for support reading n writing NDEF for Mifare Classicc

**Reply by ldaponte** [reply]
@MatthewJeffson would you offer any pointers to getting NFC read+write working with Arduino and the Seeed Sense (NRF52 based board)?

**Reply by danuw** [reply]
Would also be good to get a recommendation of antenna to order that is compatible with the Xiao's pins as well please

**Reply by MatthewJeffson** [reply]
Hi, I'm sorry, but the wiki will be changed for getting NFC functional when the new NFC library comes out. I will keep ask about the progress of the development.

As for the antenna, I can recommend you our [NFC antenna](https://www.seeedstudio.com/NFC-Antenna-p-1805.html). You can remove the wires from the solder spots on the board and replace(re-solder) them with two wires.

**Reply by danuw** [reply]
Thank you so to confirm, recommendation is:
- to buy that antenna you linked
- remove the wire and solder 2 new wires to the antenna square piece 
- solder these 2 wires to the 2 connection pads available on the XIAO devices (I assume wires can be soldered without worrying about direction? red and purple on this image https://wiki.seeedstudio.com/XIAO-BLE-Sense-NFC-Usage/#hardware-connection )

Thank you again

**Reply by MatthewJeffson** [reply]
Hi, yeah, you can solder two wires regardless the direction.

**Comment by danuw** [comment]
Hi any links for the antenna. Seeing this tutorial I thought there might have been one in the kit, but would welcome some more recommendations please. (they all seem pretty big online btw)
Thanks

**Comment by eemanon** [comment]
wow. Throwing a product on the market which functionalities can't be used. Sounds like fraud on the customer to me. At least add it to the product description that people know what they can expect from the products. No offense, the form and cost factor of your products is great. But the basic management of software releases is just beyond horrible.

**Reply by MatthewJeffson** [reply]
The NFC function is just temporarily not functioning. And yeah making a statement in the wiki sounds a good idea! Thank you!

**Comment by Aviortheking** [comment]
Hi, I am in need of the NFC

Is there any goal date for the release of the NFC library ?

**Reply by MatthewJeffson** [reply]
Hi, I am terribly sorry that I don't know the exact date of releasing the NFC library.
